android/cli/commands/_build.js

```diff
@@ -1472,12 +1472,24 @@ AndroidBuilder.prototype.validate = function validate(logger, config, cli) {
 				if (module.platform.indexOf('commonjs') != -1) {
 					module.native = false;
 
+					// Look for legacy module.id.js first
 					module.libFile = path.join(module.modulePath, module.id + '.js');
 					if (!fs.existsSync(module.libFile)) {
-						this.logger.error(__('Module %s version %s is missing module file: %s', module.id.cyan, (module.manifest.version || 'latest').cyan, module.libFile.cyan) + '\n');
-						process.exit(1);
+						// then package.json TODO Verify the main property points at reale file under the module!
+						module.libFile = path.join(module.modulePath, 'package.json');
+						if (!fs.existsSync(module.libFile)) {
+							// then index.js
+							module.libFile = path.join(module.modulePath, 'index.js');
+							if (!fs.existsSync(module.libFile)) {
+								// then index.json
+								module.libFile = path.join(module.modulePath, 'index.json');
+								if (!fs.existsSync(module.libFile)) {
+									this.logger.error(__('Module %s version %s is missing module files: %s, package.json, index.js, or index.json', module.id.cyan, (module.manifest.version || 'latest').cyan, path.join(module.modulePath, module.id + '.js').cyan) + '\n');
```

Comment:
Where does Node.js's implementation say that it treats `package.json` as a module? It is meant to parsed and use the `main` property to locate the real module.